File: internet-time-machine/frontend/types/domain.ts
export interface Snapshot {
  wayback_ts: string;
  captured_at: string;
  wayback_url: string;
  page_title: string | null;
}

export interface DomainMetadata {
  domain: string;
  first_captured: string | null;
  last_captured: string | null;
  total_snapshots: number;
  has_biography: boolean;
  snapshots: Snapshot[];
}

export interface DesignEra {
  name: string;
  start: string;
  end: string;
  description: string;
}

export interface KeyMoment {
  date: string;
  title: string;
  description: string;
}

export interface Biography {
  biography_md: string;
  design_eras: DesignEra[];
  key_moments: KeyMoment[];
  one_liner: string | null;
  generated_at: string;
  domain?: string;
}

export interface SavedDomain {
  id: string;
  domain: string;
  personal_note: string | null;
  saved_at: string;
  first_captured: string | null;
  latest_snapshot_ts: string | null;
}

File: internet-time-machine/backend/app/schemas/domain.py
from pydantic import BaseModel
from datetime import date, datetime
from typing import List, Optional

class SnapshotSchema(BaseModel):
    wayback_ts: str
    captured_at: datetime
    wayback_url: str
    page_title: Optional[str] = None

    class Config:
        from_attributes = True

class DomainResponse(BaseModel):
    domain: str
    first_captured: Optional[date] = None
    last_captured: Optional[date] = None
    total_snapshots: int = 0
    has_biography: bool = False
    snapshots: List[SnapshotSchema] = []

    class Config:
        from_attributes = True
